src/CodeGen_Hexagon.cpp

```diff
@@ -1395,51 +1422,10 @@ void CodeGen_Hexagon::visit(const Min *op) {
 }
 
 void CodeGen_Hexagon::visit(const Select *op) {
-    if (op->type.is_vector() && op->condition.type().is_vector()) {
-        // eliminate_bool_vectors has replaced all boolean vectors
-        // with integer vectors of the appropriate size, and this
-        // condition is of the form 'cond != 0'. We just need to grab
-        // cond and use that as the operand for vmux.
-        Expr cond = op->condition;
-        const NE *cond_ne_0 = cond.as<NE>();
-        if (cond_ne_0) {
-            internal_assert(is_zero(cond_ne_0->b));
-            cond = cond_ne_0->a;
-        }
-        Expr t = op->true_value;
-        Expr f = op->false_value;
-        value = call_intrin(op->type,
-                            "halide.hexagon.mux" + type_suffix(t, f, false),
-                            {cond, t, f});
-    } else if (op->type.is_vector()) {
-        // Implement scalar conditions with if-then-else.
-        BasicBlock *true_bb = BasicBlock::Create(*context, "true_bb", function);
-        BasicBlock *false_bb = BasicBlock::Create(*context, "false_bb", function);
-        BasicBlock *after_bb = BasicBlock::Create(*context, "after_bb", function);
-        builder->CreateCondBr(codegen(op->condition), true_bb, false_bb);
-
-        builder->SetInsertPoint(true_bb);
-        Value *true_value = codegen(op->true_value);
-        builder->CreateBr(after_bb);
-        // The true value might have jumped to a new block (e.g. if there was a
-        // nested select). To generate a correct PHI node, grab the current
-        // block.
-        BasicBlock *true_pred = builder->GetInsertBlock();
-
-        builder->SetInsertPoint(false_bb);
-        Value *false_value = codegen(op->false_value);
-        builder->CreateBr(after_bb);
-        BasicBlock *false_pred = builder->GetInsertBlock();
-
-        builder->SetInsertPoint(after_bb);
-        PHINode *phi = builder->CreatePHI(true_value->getType(), 2);
-        phi->addIncoming(true_value, true_pred);
-        phi->addIncoming(false_value, false_pred);
-
-        value = phi;
-    } else {
-        CodeGen_Posix::visit(op);
-    }
+    internal_assert(op->condition.type().is_scalar());
+
+    // Implement scalar conditions with if-then-else.
+    codegen(Call::make(op->type, Call::if_then_else, {op->condition, op->true_value, op->false_value}, Call::Intrinsic));
```

Comment:
(In general, it would be nice to minimize divergence from CodeGen_LLVM, and over time, hopefully we can remove more divergence.)